pkg/domain/domain.go

```diff
@@ -1808,6 +1810,25 @@ func (do *Domain) GetPDHTTPClient() pdhttp.Client {
 	return nil
 }
 
+func decodePrivilegeEvent(resp clientv3.WatchResponse) PrivilegeEvent {
+	var event PrivilegeEvent
+	for _, event := range resp.Events {
+		if event.Kv != nil {
+			val := event.Kv.Value
+			if len(val) > 0 {
+				err := json.Unmarshal(val, &event)
+				if err == nil {
+					break
+				}
+			}
+		}
+	}
+	if len(event.UserList) == 0 {
+		event.All = true
+	}
```

Comment:
set the `All` to true at event producer side, we only need to consume the event

Reply:
I do it here on purpose.
One consideration is the old tidb compatibility.
The old tidb will not set any user list, so the json message is null.
And the default behavior before is reload all ...

So this can keep the compatibility with the old tidb.... old tidb notify to new tidb cause a reload all behaviour.

@D3Hunter

Reply:
can you put this in the comment, else we don't know the context for this branch

Reply:
i see there is a `if len(val) > 0 {` branch above, maybe set it there?

Reply:
Done